core/src/main/scala/slamdata/AutoBump.scala

```diff
@@ -0,0 +1,369 @@
+/*
+ * Copyright 2020 Precog Data
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package precog
+import java.io.File
+import java.nio.file.Files
+
+import org.http4s.Uri
+
+import cats.Monad
+import cats.data.EitherT
+import cats.effect.IO.contextShift
+import cats.effect.{ContextShift, IO, Sync}
+import cats.implicits._
+import github4s.GithubResponses.{GHException, GHResponse, GHResult}
+import github4s.domain._
+import precog.domain.{PullRequestDraft, PullRequestUpdate}
+import sbt.{Logger, url}
+import sbttrickle.metadata.OutdatedRepository
+import fs2.{Chunk, Stream}
+
+import scala.collection.immutable.Seq
+import scala.collection.mutable
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.sys.process._
+import scala.util.Try
+import scala.util.matching.Regex
+
+object AutoBump {
+  sealed trait Warnings {
```

Comment:
When encoding an ADT, it's usually best to make the parent type `extend Product with Serializable` so that your type errors don't look horrible.